src/main/psxsdk/libspu/s_ini.c

```diff
@@ -1,7 +1,59 @@
 #include "common.h"
 #include "libspu_internal.h"
 
-INCLUDE_ASM("main/nonmatchings/psxsdk/libspu/s_ini", _SpuInit);
+extern s32 _spu_fd;
+extern s32 _spu_trans_mode;
+extern s32 _spu_rev_flag;
+extern s32 _spu_rev_reserve_wa;
+extern s32 _spu_rev_offsetaddr;
+
+struct SpuRevAttr {
+    s32 unk0;
+    s32 unk18;
+    s16 unk1c;
+    s16 unk1e;
+    s32 unk20;
+    s32 unk24;
+};
+
+extern struct SpuRevAttr _spu_rev_attr;
+
+extern s16 _spu_voice_centerNote[];
+
+extern s32 _spu_EVdma;
+
+extern s32 _spu_keystat;
+
+extern s32 _spu_rev_startaddr[];
+
+void _SpuInit(s32 arg0) {
+    u16* var_v0;
+    s32 var_v1;
+    u16 temp;
+
+    ResetCallback();
+    _spu_init(arg0);
+    if (arg0 == 0) {
+        temp = 0xC000;
+        for (var_v1 = 0x17, var_v0 = &_spu_voice_centerNote[0x17]; var_v1 >= 0;
```

Comment:
`_spu_voice_centerNote` seems to be a fake symbol. I re-worked the for loop in a different way which I believe it is closer to how the original code would have looked like: https://decomp.me/scratch/F3iqN

Reply:
This object is fully matched https://github.com/sozud/psy-q-decomp/blob/master/src/spu/s_ini.c
The symbol is at offset 40 (0x28) so I believe it's right. The loop change matches in the original so I'll fix that.
```
section .data symbol name _spu_voice_centerNote offset 40
```